packages/devtools_app/test/code_size_screen_test.dart

```diff
@@ -176,35 +220,88 @@ void main() {
       expect(
-        find.byKey(CodeSizeScreen.snapshotViewTreemapKey),
+        find.byKey(CodeSizeScreen.diffViewTreemapKey),
         findsOneWidget,
       );
+      expect(
+        find.byKey(CodeSizeScreen.snapshotViewTreemapKey),
+        findsNothing,
+      );
 
-      // Assumes the treemap is built with treemap_test_data_v8_new.json and treemap_test_data_v8_old.json
-      const text = 'package:pointycastle';
-      expect(find.text(text), findsOneWidget);
-      await tester.tap(find.text(text));
-      await tester.pumpAndSettle();
-
-      expect(find.text('ecc'), findsOneWidget);
-      expect(find.text('dart:core'), findsNothing);
+      expect(find.text('Root [+1.5 MB]'), findsOneWidget);
+      expect(find.text('package:pointycastle'), findsOneWidget);
+      expect(find.text('package:flutter'), findsOneWidget);
 
       expect(find.byType(CodeSizeSnapshotTable), findsNothing);
       expect(find.byType(CodeSizeDiffTable), findsOneWidget);
     });
+
+    testWidgetsWithWindowSize(
+        'loads data and shows different tree types', windowSize,
+        (WidgetTester tester) async {
+      await loadDiffTabAndSettle(tester);
+
+      codeSizeController.loadDiffTreeFromJsonFiles(
+        oldV8JsonFile,
+        newV8JsonFile,
+        (error) => {},
+      );
+      await tester.pumpAndSettle();
+
+      await tester.tap(find.byKey(CodeSizeScreen.dropdownKey));
+      await tester.pumpAndSettle();
+
+      await tester.tap(find.text('Increase Only').hitTestable());
+      await tester.pumpAndSettle();
+
+      expect(find.text('Root [+1.6 MB]'), findsOneWidget);
+      expect(find.text('package:pointycastle'), findsOneWidget);
+      expect(find.text('package:flutter'), findsOneWidget);
+
+      await tester.tap(find.byKey(CodeSizeScreen.dropdownKey));
+      await tester.pumpAndSettle();
+
+      await tester.tap(find.text('Decrease Only').hitTestable());
+      await tester.pumpAndSettle();
+
+      expect(find.text('Root'), findsOneWidget);
+      expect(find.text('package:memory'), findsOneWidget);
+      expect(find.text('package:flutter'), findsOneWidget);
+    });
+
+    testWidgetsWithWindowSize('clears data', windowSize,
+        (WidgetTester tester) async {
+      await loadDiffTabAndSettle(tester);
+
+      codeSizeController.loadDiffTreeFromJsonFiles(
+        oldV8JsonFile,
+        newV8JsonFile,
+        (error) => {},
+      );
+      await tester.pumpAndSettle();
+
+      await tester.tap(find.byKey(CodeSizeScreen.clearButtonKey));
+      await tester.pumpAndSettle();
+
+      expect(find.byType(DualFileImportContainer), findsOneWidget);
+      expect(find.byType(FileImportContainer), findsNWidgets(2));
+      expect(find.text(DiffView.importOldInstructions), findsOneWidget);
+      expect(find.text(DiffView.importNewInstructions), findsOneWidget);
+      expect(find.text('No File Selected'), findsNWidgets(2));
+    });
   });
 
-  group('CodeSizeController', () {
+  group('CodeSizeTestController', () {
```

Comment:
we can leave this as CodeSizeController

Reply:
Do you mean for any instance where we don't need to use delay? So we'd just use it for the case where we need to use the overridden `loadTreeFromJsonFile` with a delay.

Reply:
I just mean there is no need to rename the testing group - CodeSizeTestController is just an abstraction to make it easier for us to test

Reply:
Ah, sorry I didn't catch that at all haha, thanks!